lib/Ed25519Signature2020.js

```diff
@@ -107,19 +110,27 @@ export class Ed25519Signature2020 extends LinkedDataSignature {
   }
 
   async assertVerificationMethod({verificationMethod}) {
+    let contextUrl;
+    let requiredKeyType;
+    if(verificationMethod.type === 'Ed25519VerificationKey2018') {
+      contextUrl = SUITE_CONTEXT_URL_2018;
+      requiredKeyType = 'Ed25519VerificationKey2018';
+    } else if(verificationMethod.type === 'Ed25519VerificationKey2020') {
+      contextUrl = SUITE_CONTEXT_URL;
+      requiredKeyType = 'Ed25519VerificationKey2020';
+    }
     if(!_includesContext({
-      document: verificationMethod, contextUrl: SUITE_CONTEXT_URL
+      document: verificationMethod, contextUrl
     })) {
       // For DID Documents, since keys do not have their own contexts,
       // the suite context is usually provided by the documentLoader logic
       throw new TypeError(
-        `The verification method (key) must contain "${SUITE_CONTEXT_URL}".`
+        `The verification method (key) must contain "${contextUrl}".`
       );
     }
-
-    if(!jsonld.hasValue(verificationMethod, 'type', this.requiredKeyType)) {
+    if(!jsonld.hasValue(verificationMethod, 'type', requiredKeyType)) {
```

Comment:
This is a bit confusing because this is the very same assertion being made in the new conditionals on L115.  Is there some way to consolidate this?  Or can this assertion/test be eliminated?

Reply:
Removed fce8215